nomad/state/paginator.go

```diff
@@ -62,6 +87,19 @@ func (p *Paginator) next() (interface{}, paginatorState) {
 	if !p.nextTokenFound && id < p.seekingToken {
 		return nil, paginatorSkip
 	}
+
+	// apply filter if defined
+	if p.filterEvaluator != nil {
+		match, err := p.filterEvaluator.Evaluate(raw)
+		if err != nil {
+			p.pageErr = err
+			return nil, paginatorComplete
+		}
```

Comment:
@tgross any thoughts on this approach for handling errors during pagination? 

I thought of creating a new `paginatorState` called `paginatorError`, but having a non-nil `pageErr` would already indicate that so I just used the `paginatorComplete` state to break the loop.

Reply:
This is fine as-is, but if you want to go the extra mile and make this really conventional Go, take a look at the `Err()` pattern used for `bufio.Scanner`

https://pkg.go.dev/bufio#example-Scanner-Lines

The caller then only deals with 1 error after iteration is "complete", rather than on each iteration.

Reply:
Ah interesting, good to know about this pattern. But I think that, in this case, callers would deal with one page at a time, so having to remember to call `Err()` may be a little more error-prone?